homeassistant/components/climate/melissa.py

```diff
@@ -0,0 +1,274 @@
+        if self._cur_settings is not None:
+            return self.melissa_op_to_hass(
+                self._cur_settings[self._api.MODE])
+        else:
+            _LOGGER.info(
+                "Can't determine current operation mode of %s", self.entity_id)
+            return STATE_UNKNOWN
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return [
+            STATE_AUTO, STATE_HEAT, STATE_COOL, STATE_DRY, STATE_FAN_ONLY
+        ]
+
+    @property
+    def fan_list(self):
+        """List of available fan modes."""
+        return [
+            STATE_AUTO, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH
+        ]
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        if self._cur_settings is not None:
+            return self._cur_settings[self._api.TEMP]
+        else:
+            _LOGGER.info(
+                "Can not determine current target temperature for %s",
+                self.entity_id)
+            return STATE_UNKNOWN
+
+    @property
+    def state(self):
+        """Return current state."""
+        if self._cur_settings is not None:
+            return self.melissa_state_to_hass(
+                self._cur_settings[self._api.STATE])
+        else:
+            _LOGGER.info("Cant determine current state for %s", self.entity_id)
+            return STATE_UNKNOWN
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement which this thermostat uses."""
+        return TEMP_CELSIUS
+
+    @property
+    def min_temp(self):
+        """Return the minimum supported temperature for the thermostat."""
+        return 16
+
+    @property
+    def max_temp(self):
+        """Return the maximum supported temperature for the thermostat."""
+        return 30
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        return self.send({self._api.TEMP: temp})
+
+    def set_fan_mode(self, fan):
+        """Set fan mode."""
+        fan_mode = self.hass_fan_to_melissa(fan)
+        return self.send({self._api.FAN: fan_mode})
+
+    def set_operation_mode(self, operation_mode):
+        """Set operation mode."""
+        mode = self.hass_mode_to_melissa(operation_mode)
+        return self.send({self._api.MODE: mode})
+
+    def turn_on(self):
+        """Turn on device."""
+        return self.send({self._api.STATE: self._api.STATE_ON})
+
+    def turn_off(self):
+        """Turn off device."""
+        return self.send({self._api.STATE: self._api.STATE_OFF})
+
+    def send(self, value):
+        """Sending action to service."""
+        try:
+            old_value = self._cur_settings.copy()
+            self._cur_settings.update(value)
+        except AttributeError:
```

Comment:
When can this error happen?

Reply:
I think it's when it have not yet got the status from Melissa... Not sure atm. really.